paper.tex

```diff
@@ -228,33 +250,127 @@ \subsection*{High-level model specification in YAML}
+
+\jkcomment{Finish up this section with a discussion of how we (a)
+use default values defined in JSON schema; (b) define more
+subtle implicit values in the specification itself; and (c) use the
+defaults sections, all explified by Fig 1.}
+
+\subsection*{Parsers}
+The implicit value inference and default value propagation required to
+fully resolve the high-level Demes model description is straightforward
+to describe, but still requires significant effort to implement in
+software. Moreover, there are many constraints on the data model
+(for example, epoch \texttt{end\_time} values should be strictly increasing
+within a deme), which must be enforced.
+It would not be reasonable to require every program that
+takes Demes models as input to implement this logic, as the programming
+effort would be significant (limiting adoption)
+and it is likely that if there were many implementations some would differ
+in their details (harming the software ecosystem). We therefore define
+a standard software entity as part of the specification (the parser),
+which performs this task, and which can be shared by programs that
+support demes as an input. By having relatively few (ideally, one
+per programming language), high-quality Demes parsers available as
+libraries the probability of divergences from the standard is
+greatly reduced.
+
+The Demes specification precisely defines the required behaviour of
+parsers, and we also provide a reference implementation written
+in Python to resolve any potential ambiguities and to provide a
+helpful template for other implementations. We also provide an
+extensive suite suite of examples and the expected outputs. In addition,
+we have high-quality parser implementations in the Python and C
+languages (published under liberal open source licences),
+providing a solid foundation for the software ecosystem.
 
-\subsection*{Design rationale}
 
-This is an unstructured list of things we should mention in terms
-of why certain decisions were made
-
-\begin{itemize}
-    \item Recent developments in software engineering have favoured
-    static, declarative configuration formats. It feels like a good idea to
-    have a Turing complete configuration format, but it turns out not to be.
+\subsection*{Unambiguous low-level representation}
 
-    \item Drawing parameters from distributions. Why we are static.
-    % https://github.com/popsim-consortium/demes-paper/issues/3
+\jkcomment{Not sure if there's much point in this as a separate section any
+more? This could really be said in the previous section.}
+
+Parsers output the low-level description of Demes models,
+which is an explicit representation
+of the model where all parameters have been assigned values, and
+all data constraints have been checked. The output is formally defined
+as JSON, but in practise parsers output an object model that
+is suitable for the particular programming language, and
+that can be used directly by the implementing program.
```

Comment:
I get what you're saying here, but it might be better to explicitly say that parsers can construct class instances in an OO language, or structs in C, etc. as the parser's "output".